Core/Core/Todos/Todo.swift

```diff
@@ -59,12 +120,23 @@ public final class Todo: NSManagedObject, WriteableModel {
     }
 
     public static func save(_ item: APITodo, in context: NSManagedObjectContext) -> Todo {
-        let id = item.assignment.id.value
-        let assignment: Assignment = context.first(where: (\Assignment.id).string, equals: id) ?? context.insert()
-        assignment.update(fromApiModel: item.assignment, in: context, updateSubmission: false, updateScoreStatistics: false)
-
+        let id = item.assignment?.id.value ?? item.quiz?.id.value ?? UUID.string
         let model: Todo = context.first(where: #keyPath(Todo.id), equals: id) ?? context.insert()
-        model.assignment = assignment
+
+        if let apiAssignment = item.assignment {
+            let assignment: Assignment = context.first(where: (\Assignment.id).string, equals: id) ?? context.insert()
+            assignment.update(fromApiModel: apiAssignment, in: context, updateSubmission: false, updateScoreStatistics: false)
+            model.assignment = assignment
+            model.name = assignment.name
+            model.dueAtSortNilsAtBottom = assignment.dueAt ?? .distantFuture
+        } else if let apiQuiz = item.quiz {
+            let quiz: Quiz = Quiz.save(apiQuiz, in: context)
+            quiz.courseID = item.course_id?.value ?? "unknown"
+            model.quiz = quiz
+            model.name = quiz.title
+            model.dueAtSortNilsAtBottom = quiz.dueAt ?? .distantFuture
```

Comment:
Just to double check: these are full quiz and assignment objects and not missing any fields that could be reset from the DB while saving these entities?

Reply:
I double checked the quiz api they are full objects. The assignment fetching didn't change with this PR but I noticed the `assignment_groups` api doesn't return assignments with due date. Not sure if this is intentional API design but it's the same in the app store version.